### PR TITLE
Task 6 with tests and endpoints

### PR DESCRIPTION
**Pull request by asazycat**
I filled in the endpoints, made the tests and passed them. I made sure to leave comments and describe.onlys out ,or at I hope so.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -163,8 +163,30 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
+               expect(eachComment).toHaveProperty("comment_id", expect.any(Number))
+               expect(eachComment).toHaveProperty("votes", expect.any(Number)) 
+               expect(eachComment).toHaveProperty("created_at", expect.any(String))
+               expect(eachComment).toHaveProperty("author", expect.any(String))
+                 
+               expect(eachComment).toHaveProperty("body", expect.any(String))
+               expect(eachComment).toHaveProperty("article_id", expect.any(Number))
+                       })
+
+                       expect(body.comments).toBeSorted();
```
Comment (itskatherine) [inline __tests__/app.test.js:185]: This expect statement should specify what the comments are sorted by - i.e. `created_at` - jest-sorted has functionality that can do this 🔍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -163,8 +163,30 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
 
 
-
-
+            describe('gets an array of comments back for the specified id', ()=> {
+                test('check if the array has been returned with comments', ()=> {
+   
+                 return  request(app).get('/api/articles/3/comments').expect(200).then(({body})=>{
+   
+                       expect(body.comments.length).toBe(2)
```
Comment (itskatherine) [inline __tests__/app.test.js:171]: Nice length check to avoid false positives in the `forEach` 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -163,8 +163,30 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
+   
+                       body.comments.forEach((eachComment) => {
+   
+               expect(eachComment).toHaveProperty("comment_id", expect.any(Number))
+               expect(eachComment).toHaveProperty("votes", expect.any(Number)) 
+               expect(eachComment).toHaveProperty("created_at", expect.any(String))
+               expect(eachComment).toHaveProperty("author", expect.any(String))
+                 
+               expect(eachComment).toHaveProperty("body", expect.any(String))
+               expect(eachComment).toHaveProperty("article_id", expect.any(Number))
```
Comment (itskatherine) [inline __tests__/app.test.js:182]: For the comments coming back from this endpoint, wouldn't we always know the exact value of `article_id`? 🕵️

File: __tests__/app.test.js [hunk 4]
```diff
@@ -163,8 +163,30 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
+                       })
+
+                       expect(body.comments).toBeSorted();
+                   })
+                })
+                     })
+
```
Comment (itskatherine) [inline __tests__/app.test.js:189]: A solid happy path test ✅ Missing a bit of error testing here though 📝, we need to check:
- we get 400 when an invalid datatype is provided as the`article_id`, (`/api/articles/katherinerules/comments`)
- we get 404 when a valid but non-existant `article_id` is provided (`/api/articles/999999/comments`)
- we get 200 and an empty array when we ask for the comments from a valid existing article that has no comments (i.e. this scenario shouldn't trigger an error)

File: controller/app.controller.js [hunk 5]
```diff
@@ -45,4 +45,18 @@ exports.getArticles = (req,res,next) => {
  
     res.status(200).send({articles:allArticlesArray})
   }).catch(next)
+}
+
+
+
+exports.getCommentsByArticleId = (req,res,next) => {
+  const {article_id} = req.params
+
```
Comment (itskatherine) [inline controller/app.controller.js:54]: Not mandatory but I'd recommend getting a code formatter like Prettier, to improve this white space everywhere and make your code a bit easier to read 👓

File: endpoints.json [hunk 6]
```diff
@@ -25,5 +25,43 @@
...
+    },
+    {
+      "body": "Ambidextrous marsupial",
+      "votes": 0,
+      "author": "icellusedkars",
+      "article_id": 3,
+      "created_at": 1600560600000
+    }]
+
+
```
Comment (itskatherine) [inline endpoints.json:64]: 👍

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Asad, a solid start! But need to cover a few more test scenarios
Have a look over my comments and once you've addressed them can you please submit another PR 👍

File: __tests__/app.test.js [hunk 7]
```diff
@@ -163,11 +163,69 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
+               expect(eachComment).toHaveProperty("comment_id", expect.any(Number))
+               expect(eachComment).toHaveProperty("votes", expect.any(Number)) 
+               expect(eachComment).toHaveProperty("created_at", expect.any(String))
+               expect(eachComment).toHaveProperty("author", expect.any(String))
+                 
+               expect(eachComment).toHaveProperty("body", expect.any(String))
+               expect(eachComment).toHaveProperty("article_id", expect.any(Number))
+                       })
+
+                       expect(body.comments).toBeSorted();
```
Comment (lewispricey) [inline __tests__/app.test.js:185]: you may want to specify how these should be sorted, the first arguement the toBeSorted by takes will be the key that should be sorted.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -163,11 +163,69 @@ expect(body).toHaveProperty("article_id", expect.any(Number))
...
+   
+                       body.comments.forEach((eachComment) => {
+   
+               expect(eachComment).toHaveProperty("comment_id", expect.any(Number))
+               expect(eachComment).toHaveProperty("votes", expect.any(Number)) 
+               expect(eachComment).toHaveProperty("created_at", expect.any(String))
+               expect(eachComment).toHaveProperty("author", expect.any(String))
+                 
+               expect(eachComment).toHaveProperty("body", expect.any(String))
+               expect(eachComment).toHaveProperty("article_id", expect.any(Number))
```
Comment (lewispricey) [inline __tests__/app.test.js:182]: Can we be specific about what ID we'd expect here

**Review by lewispricey — APPROVED** [review]
Great work on this Asad, thanks for adding in the extra tests Katherine suggested.

I've left notes of a couple of small changes to make in your tests, once these are updated please feel free to merge 🚀 

No need to submit another pull request for this ticket.

**Pull request merged**